bolt-sidecar/benches/score_cache.rs

```diff
@@ -0,0 +1,81 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::collections::HashMap;
+
+use bolt_sidecar::common::score_cache::ScoreCache;
+
+const GET_SCORE: isize = 1;
+const INSERT_SCORE: isize = 2;
+const UPDATE_SCORE: isize = 3;
+
+fn bench_scorecache_vs_hashmap(c: &mut Criterion) {
+    let mut group = c.benchmark_group("ScoreCache vs HashMap");
+
+    let mut score_cache: ScoreCache<GET_SCORE, INSERT_SCORE, UPDATE_SCORE, usize, usize> =
+        ScoreCache::with_max_len(10_000);
+    let mut hash_map: HashMap<usize, usize> = HashMap::new();
+
+    // Insert benchmark
+    group.bench_function("ScoreCache Insert", |b| {
+        b.iter(|| {
+            for i in 0..1000 {
+                score_cache.insert(i, i);
```

Comment:
Not sure if you need it here, but verify if you need to use `black_box` or not to avoid compiler to optimize